src/test/ui/intrinsics/panic-uninitialized-zeroed.rs

```diff
@@ -284,25 +284,23 @@ fn main() {
             "attempted to leave type `[core::ptr::non_null::NonNull<()>; 1]` uninitialized, which is invalid"
         );
 
-        test_strict_panic_msg(
-            || mem::zeroed::<LR_NonZero>(),
-            "attempted to zero-initialize type `LR_NonZero`, which is invalid"
+        test_panic_msg(
+            || mem::uninitialized::<(&[u8], &str, &())>(),
+            "attempted to leave type `(&[u8], &str, &())` uninitialized, which is invalid"
```

Comment:
This one we could accept, if we wanted, by accepting all references to *sized* align-1-ZST... doesn't seem very important though.

Reply:
Though allowing `&[u8]` and rejecting `&()` feels rather arbitrary so we should probably allow both then?

Reply:
Could do, but I figure since not many people are actually making a reference to a ZST, no harm in forbidding it now.

The changes here *are* rather arbitrary in their nature, we're only not panicking on the bits that we can't panic on just yet, while making the checks stricter where we can.

Making an uninit `[bool; 1]` will start to panic after this change, even if that's still not LLVM UB if we 0x01 fill, because there's not really a widely used crate that does that (inb4 some crate started doing that since the last crater run :p).

Hyper doesn't make uninit sized ZSTs, so I don't think we need to allow them here, even if it is somewhat inconsistent to forbid them.

I'm fine with ignoring &() as well, I just don't feel it's needed.